sdk/core/azure-core/test/ut/request_conditions.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/datetime.hpp>
+#include <azure/core/etag.hpp>
+#include <azure/core/request_conditions.hpp>
+#include <gtest/gtest.h>
+#include <string>
+#include <vector>
+
+using namespace Azure::Core;
+
+TEST(RequestConditions, Basic)
+{
+  RequestConditions conditions;
```

Comment:
Where do we expect to use this, and verify the functional correctness that conditions like `IfModifiedSince` actually work as expected?